XenAdmin/Controls/CustomDataGraph/GraphHelpers.cs

```diff
@@ -192,6 +193,16 @@ public static List<DataSourceItem> BuildList(IXenObject xenObject, List<Data_sou
                 dataSourceItems.Add(new DataSourceItem(dataSource, friendlyName, Palette.GetColour(itemUuid), itemUuid, xenObject));
             }
 
+            // Filter old datasources only if we have their replacement ones
+            Regex io_throughput_rw_regex = new Regex("^io_throughput_(read|write)_([a-f0-9]{8})$"); // old datasources

```

Comment:
I think these Regex objects could be moved outside this function so it won't get created each time this is called.